### PR TITLE
Add Artifactory plugin support

### PR DESCRIPTION
**Pull request by wojtekk**
Hi, I'm really happy that I've found your plugin to Semantic Release.

In my company, we are using Artifactory. Could be possible to add support of it to your tool?

I've prepared implementation in this PR.

Best regards,
wojtekk

### CODE REVIEW

File: test/project/with-artifactory-plugin/build.gradle [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// https://www.jfrog.com/confluence/display/RTF/Gradle+Artifactory+Plugin
+plugins {
+    id "com.jfrog.artifactory" version "4.9.6"
+    id 'maven-publish'
```
Comment (KengoTODA) [inline test/project/with-artifactory-plugin/build.gradle]: I think you need to remote this `maven-publish` plugin

File: src/gradle.ts [hunk 2]
```diff
@@ -43,11 +43,20 @@ export function getTaskToPublish(
     } else {
       let task = "";
       child.stdout.pipe(split()).on("data", (line: string) => {
-        if (line.startsWith("publish -")) {
+        if (line.startsWith("artifactoryDeploy -")) {
           if (task !== "") {
```
Comment (KengoTODA) [inline src/gradle.ts]: Are we sure that `publish` always come earlier than `artifactoryDeploy`? If not, it is necessary to add if-block here.

```js
        if (line.startsWith("artifactoryDeploy -")) {
          if (task !== "" && task !== "publish") {
            reject(new Error("Found multiple tasks to publish"));
          }
          if (task === "publish") {
            // TODO: log to say 'we found both artifactory and maven-publish so we use artifactory for release'
          }
          task = "artifactoryDeploy";
```

**Comment by wojtekk** [comment]
Thank you for fast feedback. You are right in both cases:

- [x] I'm not 100% sure about tasks order
- [x] I don't have to import `maven-publish` plugin in `build.gradle` but I need to enable it to be sure that the task `publish` is available in the test (the key point that `maven-publish` plugin is often used with `artifactory-publish` i.e. [question on stackoverflow](https://stackoverflow.com/questions/22352475/upload-artifact-to-artifactory-using-gradle) so I want to cover that in the test).

**Comment by KengoTODA** [comment]
Thanks @wojtekk, I've merged this PR! :tada:

**Comment by wojtekk** [comment]
Thank you!

**Pull request merged**